sw/device/sca/lib/sca.c

```diff
@@ -337,6 +337,35 @@ void sca_call_and_sleep(sca_callee callee, uint32_t sleep_cycles) {
       &clkmgr, CLKMGR_CLK_ENABLES_CLK_IO_DIV4_PERI_EN_BIT, kDifToggleEnabled));
 }
 
+void sca_call_and_sleep_trigger(sca_callee callee, uint32_t sleep_cycles) {
+  // Disable the IO_DIV4_PERI clock to reduce noise during the actual capture.
+  // This also disables the UART(s) and GPIO modules required for
+  // communication with the scope. Therefore, it has to be re-enabled after
+  // the capture.
+  dif_clkmgr_t clkmgr;
+  OT_DISCARD(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  // Start timer to wake Ibex after the callee is done.
+  uint64_t current_time;
+  // Return values of below functions are ignored to improve capture
+  // performance.
+  OT_DISCARD(dif_rv_timer_counter_read(&timer, kRvTimerHart, &current_time));
+  OT_DISCARD(dif_rv_timer_arm(&timer, kRvTimerHart, kRvTimerComparator,
+                              current_time + sleep_cycles));
+  OT_DISCARD(dif_rv_timer_counter_set_enabled(&timer, kRvTimerHart,
+                                              kDifToggleEnabled));
+  sca_set_trigger_high();
```

Comment:
As far as I can tell, this function is identical to `sca_call_and_sleep()` except for the calls to the trigger functions. To reduce maintenance overhead, would it be better to extend the existing function with an argument to decide whether the trigger needs to be handled? Or would the branch this involves negatively impact the measurement?

Reply:
Thanks for the comment. Done!

Reply:
Nice, thanks!